### PR TITLE
feature/117 - restrict ES highlighting to included fields unless context.showOtherMatches

### LINKED ISSUE #117: Restrict highlighting to includes if results node doesn't have showOtherMatches: true
We're hiding the other matches column in contexture-react unless the results node has `showOtherMatches: true`: https://github.com/smartprocure/contexture-react/issues/163. 

In this repo, we need to restrict highlighting on un-displayed fields unless we've got that showOtherMatches flag. This will save ES the work of highlighting extra results fields that won't get shown to the user.

### PR DESCRIPTION
**Pull request by doug-patterson**
Fixes #117 

### Description
* restricts highlighting to context.includes unless context.showOtherMatches is set truthy

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please assign someone to merge this PR, and optionally include people who should review.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The changelog has not been updated. Please update the changelog.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The version was not updated. Please update the version.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  You should add at least 1 more reviewer to the PR
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  We were able to automatically fix some formatting issues in this PR for you!
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by doug-patterson** [comment]
actually, now that I've looked at the unit tests for the results component they're already broken two ways

* `resultsTest` isn't awaited so the tests all "pass" and instead you get complaints about an un-handled promise rejection
* the first test for the `results` type is already failing.

I'll need to fix those things and then write new tests for `showOtherMatches`.

**Comment by doug-patterson** [comment]
@sadasant here's the output if you fix the `await` problem at https://github.com/smartprocure/contexture-elasticsearch/blob/master/test/example-types/results.js#L74 and then run the tests:

```
  144 passing (196ms)
  1 failing

  1) results should be able to filter fields with include:

      AssertionError: expected { Object (scrollId, response) } to deeply equal { Object (scrollId, response) }
      + expected - actual

             {
               "_id": "test-id"
               "additionalFields": []
               "field": "test field"
      -        "hit": {
      -          "_id": "test-id"
      -          "field": "test field"
      -        }
             }
           ]
           "startRecord": 1
           "totalRecords": 1
      
      at _.curry (test/example-types/testUtils.js:47:28)

```

I see you primarily in the blame for the test so I figure I'll ask you: what should be changed to make this test both accurate and passing? Since I'm new to the repo I want to be sure I don't "fix" the test that's already broken incorrectly

**Comment by doug-patterson** [comment]
@dubiousdavid @dshishkov @sadasant @Akrion need a review on this for https://github.com/smartprocure/bid-search/issues/1125

File: test/example-types/results.js [hunk 1]
```diff
@@ -164,7 +164,7 @@ describe('results', () => {
     ])
   })
   it('should sort on "_score: desc" with no sortField config', async () =>
```
Comment (dshishkov) [inline test/example-types/results.js]: Looks like we can drop the `async` from this test since it seems to be synchronous now.

**Comment by doug-patterson** [comment]
@dshishkov updated

**Comment by doug-patterson** [comment]
@dshishkov @dubiousdavid ready for review

**Comment by doug-patterson** [comment]
@dshishkov @dubiousdavid updated, ready for review

**Pull request merged**